test/functional/apps/discover/_context_encoded_url_param.ts

```diff
@@ -32,14 +31,18 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
       });
       await PageObjects.settings.createIndexPattern('context-encoded-param');
 
-      await kibanaServer.uiSettings.update({ 'doc_table:legacy': false });
+      await kibanaServer.uiSettings.update({
+        'doc_table:legacy': false,
+        'discover:searchOnPageLoad': true,
+      });
```

Comment:
this is an interesting case since 'discover:searchOnPageLoad' should already be `true`. I'd suggest to add to the `after` part of `_search_on_page_load.ts`
https://github.com/elastic/kibana/blob/04acd494967400767774233c6afa4e122a58f25f/test/functional/apps/discover/_search_on_page_load.ts#L55

`await kibanaServer.uiSettings.replace(defaultSettings);`

So it should be cleaned up correctly , and since L34-L37 are setting default values, it could be removed.